examples/example-clientlib-ktx-app/app/src/main/java/com/solanamobile/ktxclientsample/viewmodel/SampleViewModel.kt

```diff
@@ -62,116 +69,114 @@ class SampleViewModel @Inject constructor(
-                            )
-
-                            val sig = result.payload.signatures.firstOrNull()
-                            val readableSig = Base58.encode(sig)
-
-                            _state.value.copy(
-                                isLoading = false,
-                                memoTx = readableSig
-                            ).updateViewState()
-
-                            //Clear out the recent transaction
-                            delay(5000)
-                            _state.value.copy(memoTx = "").updateViewState()
-                        }
-                        else -> {
-                            _state.value.copy(
-                                isLoading = false,
-                            ).updateViewState()
-                        }
-                    }
+                    val sig = signAndSendTransactions(arrayOf(bytes)).signatures.firstOrNull()
+                    Base58.encode(sig)
                 }
+            }.successPayload?.let { readableSig ->
+                _state.value.copy(
+                    isLoading = false,
+                    memoTx = readableSig
+                ).updateViewState()
+
+                // Clear out the recent transaction
+                delay(5000)
+                _state.value.copy(memoTx = "").updateViewState()
             }
         }
     }
 
-    private suspend fun connectIfNeeded(sender: ActivityResultSender): Boolean {
-        val conn = persistanceUseCase.getWalletConnection()
-
-        return if (conn is Connected) {
-            true
-        } else {
-            when (val result = walletAdapter.connect(sender)) {
-                is TransactionResult.Success -> {
-                    val currentConn = Connected(
-                        PublicKey(result.authResult.publicKey),
-                        result.authResult.accountLabel ?: "",
-                        result.authResult.authToken
-                    )
-
-                    persistanceUseCase.persistConnection(currentConn.publicKey, currentConn.accountLabel, currentConn.authToken)
-
-                    _state.value.copy(
-                        userAddress = currentConn.publicKey.toBase58(),
-                        userLabel = currentConn.accountLabel,
-                        solBalance = solanaRpcUseCase.getBalance(currentConn.publicKey)
-                    ).updateViewState()
-
-                    true
-                }
-                is TransactionResult.NoWalletFound -> {
-                    _state.value.copy(
-                        walletFound = false
-                    ).updateViewState()
+    fun signIn(sender: ActivityResultSender) {
+        viewModelScope.launch {
+            connect(sender) {}
+            // Note: should check the signature here of the signInResult to verify it matches the
+            // account and expected signed message. This is left as an exercise for the reader.
```

Comment:
😂

If trivial we should eventually have a snippet of this. Although true SIWS verification should happen on the server side anyways.